packages/plugin-legacy/index.js

```diff
@@ -260,7 +260,7 @@ function viteLegacyPlugin(options = {}) {
         ]
       })
 
-      return { code, map }
+      return { code: `;(function(){${code}})();`, map }
```

Comment:
@cztomsik shouldn't the sourcemap (`map`) be also updated? Using magic-string for example

Reply:
good point but I'm busy now so maybe later or somebody else can fix it properly, this is just what worked for us